applications/app/templates/serviceWorker.scala.js

```diff
@@ -8,201 +8,267 @@
+ * CACHE HELPERS
+ */
 
-                    if (allAssetResponsesOk) {
-                        return cacheOfflineCrosswordAndAssetResponses(jsonResponse, assetResponses);
-                    }
-                });
-            });
-        }
+var isCacheUpdated = function () {
+    return caches.keys().then(function (keys) {
+        return keys.some(cacheKeyMatchesTodaysCache);
     });
 };
 
 var deleteOldCaches = function () {
     return caches.keys().then(function (keys) {
         return Promise.all(
             keys.map(function (key) {
-                if (!keyMatchesTodaysCache(key)) {
+                if (!cacheKeyMatchesTodaysCache(key)) {
                     return caches.delete(key);
                 }
             })
         );
     })
 };
 
-var keyMatchesTodaysCache = function (key) {
-    return new RegExp('^' + getISODate() + '-').test(key);
+var cacheKeyMatchesTodaysCache = (function (key) {
+    var regex = new RegExp('^' + getISODate() + '-');
+    return function (key) {
+        return regex.test(key);
+    }
+})();
+
+/**
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * FETCH HELPERS
+ */
+
+var fetchAll = function (resources) {
+    return Promise.all(resources.map(function (resource) {
+        return fetch(resource);
+    }));
 };
 
-var doesRequestAcceptHtml = function (request) {
+var needCredentialsWorkaround = (function (url) {
+    var whitelistRegexs = [
+        'https://discussion.theguardian.com/discussion-api'
+    ].map(function (entry) {
+        return new RegExp('^' + entry);
+    });
+    return function (url) {
+        return whitelistRegexs.some(function (re) {
+            return re.test(url);
+        });
+    }
+})();
+
+var requestAcceptsHTML = function (request) {
     return request.headers.get('Accept')
         .split(',')
         .some(function (type) { return type === 'text/html'; });
 };
 
-var isCacheUpdated = function () {
-    return caches.keys().then(function (keys) {
-        return keys.some(keyMatchesTodaysCache);
-    });
+var isSameHost = function (host) {
+    return host === self.location.host;
 };
 
-self.addEventListener('install', function (event) {
-    event.waitUntil(updateOfflineCrosswordCache());
-});
+/**
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * FETCH HANDLERS
+ */
 
-var needCredentialsWorkaround = function (url) {
-    var whitelist = ['https://discussion.theguardian.com/discussion-api'];
-    return whitelist.some(function (entry) {
-        return new RegExp('^' + entry).test(url);
-    });
-};
+var requestForDevBlog = (function (event) {
+    var isForDevBlog = (function (url, request) {
+        var devBlogPathRegex = /^\/info\/developer-blog($|\/.*$)/;
+        return function (url, request) {
+            return isSameHost(url.host) && url.pathname.match(devBlogPathRegex) && requestAcceptsHTML(request);
```

Comment:
Minor comment but I always use `RegExp.prototype.test` (which returns `boolean`) instead of `String.prototype.match` (which returns `null` or `array`), just because you don't forget about checking the type. In this case there's not much in it though